lib/cloudwatchlogs/client.go

```diff
@@ -109,47 +110,59 @@ func openAwsClient() (client *cloudwatchlogs.CloudWatchLogs, err error) {
 	return
 }
 
-func createGroupAndStream(client *cloudwatchlogs.CloudWatchLogs, group string, stream string) (token string, err error) {
+func createGroupAndStream(client *cloudwatchlogs.CloudWatchLogs, group string, stream string) (string, error) {
 	var result *cloudwatchlogs.DescribeLogStreamsOutput
 
-	// Ignore failures on group and stream creation, describing the stream will
-	// fail later if the group doesn't exist. That way the group creation is
-	// idempotent.
-	client.CreateLogGroup(&cloudwatchlogs.CreateLogGroupInput{
+	if _, err := client.CreateLogGroup(&cloudwatchlogs.CreateLogGroupInput{
 		LogGroupName: aws.String(group),
-	})
-	client.CreateLogStream(&cloudwatchlogs.CreateLogStreamInput{
+	}); err != nil {
+		// It's OK if the log already exists
+		if err, ok := err.(awserr.Error); !ok || err.Code() != "ResourceAlreadyExistsException" {
+			// Some other error occurred
+			return "", err
+		}
+	}
+
+	_, err := client.CreateLogStream(&cloudwatchlogs.CreateLogStreamInput{
 		LogGroupName:  aws.String(group),
 		LogStreamName: aws.String(stream),
 	})
+	if err == nil {
+		// Log stream successfully created.  No token need be provided.
+		return "", nil
+	}
+
+	if err, ok := err.(awserr.Error); !ok || err.Code() != "ResourceAlreadyExistsException" {
+		// Some other error occurred
+		return "", err
+	}
```

Comment:
A construct that I often find useful to express multiple conditions that relate to each other is to use a switch/case statement, here you could make it:
```go
switch {
case err == nil:
    return "", nil
case isAlreadyExists(err):
    return "", err
}
```